python/src/iceberg/table/schema.py

```diff
@@ -0,0 +1,346 @@
+                if indexed_field_name.lower() == field_name.lower():
+                    return indexed_field_id
+        raise ValueError(f"Cannot get field ID, name not found: {field_name}")
+
+    def find_field_id_by_alias(self, field_alias: str, case_sensitive: bool = True) -> int:
+        """Get a field ID for a given field alias
+
+        Args:
+            field_alias (str): A field alias
+            case_sensitive (bool): If False, case will not be considered when retrieving the field from the schema--default is True
+
+        Returns:
+            int: The field ID for the field alias
+
+        raises:
+            ValueError: If the field ID cannot be retrieved either because the alias was not found or a case-insensitive
+                match returned multiple results
+        """
+        if case_sensitive:
+            try:
+                # For case-sensitive, just try looking up the alias and raise if it's not found
+                return self._aliases[field_alias]
+            except KeyError:
+                raise ValueError(f"Cannot get field ID, alias not found: {field_alias}")
+        if not case_sensitive:
+            matching_fields = [value for key, value in self._aliases.items() if key.lower() == field_alias.lower()]
+            if len(matching_fields) == 1:  # If one matching alias found, return the corresponding ID
+                return matching_fields[0]
+            elif not matching_fields:  # If no matching fields, raise
+                raise ValueError(f"Cannot get field ID, alias not found: {field_alias}")
+
+            # If multiple IDs are returned for a case-insensitive alias lookup, raise
+            raise ValueError(f"Cannot get field ID, case-insensitive alias returns multiple results: {field_alias}")
+
+    def find_field_name_by_field_id(self, field_id: int) -> str:
+        """Find a field name for a given field ID
+
+        Args:
+            field_id (int): A field ID
+
+        Returns:
+            str: The field name for the field ID
+
+        Raises:
+            ValueError: If the field ID does not exist
+        """
+        for indexed_field_name, indexed_field_id in self._index_by_name.items():
+            if indexed_field_id == field_id:
+                return indexed_field_name
+        raise ValueError(f"Cannot get field name, field ID not found: {field_id}")
+
+    def find_field_by_id(self, field_id: int) -> NestedField:
+        """Find a field by it's field ID
+
+        Args:
+            field_id (int): A field ID
+
+        Returns:
+            NestedField: The field object for the given field ID
+
+        Raise:
+            ValueError: If the field ID does not exist
+        """
+        for field in self.fields:
+            if field.field_id == field_id:
+                return field
+        raise ValueError(f"Cannot get field, ID does not exist: {field_id}")
+
+    def find_field_type(self, field_id: int) -> IcebergType:
```

Comment:
should this be re-named to `def find_field_type_by_id(...)` to be more explicit? 
This seems a little ambiguous 
```
table_schema.find_field_type(1). # StringType()
```

Reply:
Sounds good, updated 👍